model/pipeline/src/main/proto/beam_runner_api.proto

```diff
@@ -1206,6 +1210,11 @@ message MavenPayload {
   string repository_url = 2;
 }
 
+message DeferredArtifactPayload {
+  // A id for deferred artifacts.
+  string id = 1;
```

Comment:
I was under the impression we were going to make this a `bytes` field so that any deferred information can get passed through and then back to the **creator** whether it be an id or a serialized blob of objects or ...

Allowing for `bytes` enables for solutions beyond in memory maps.

Reply:
+1

Reply:
Don't we need at least a key field (or urn or identifier) which can be used to check whether the payload is parsable by the creator? Otherwise, it would be pretty hard to know where the bytes payload originally came from.

Reply:
Thats a good point about the key and in general this will become a problem for all artifacts since none of them have unique keys associated with them unless an intermediary resolves the artifact immediately and possibly "renames" the contents to make it unique.

If we ever want to support multiple layers of expansion for XLang we'll want to proxy any artifact resolution/retrieval calls through the layers.

Reply:
As discussed (but putting here for the record) having a proxy artifact type could solve this issue.